app/controllers/organizations/default_pet_tasks_controller.rb

```diff
@@ -1,7 +1,59 @@
 class Organizations::DefaultPetTasksController < Organizations::BaseController
```

Comment:
All looks good. Raising errors in controller actions is different from implementation in other controllers in this app but I am not opposed to it.

One thing, can we use `DefaultPetTask.new` instead of `@organization.default_pet_tasks.build` because `DefaultPetTask` has a belongs_to relatinship to Organization. So any instances will be scoped to the organization automatically thanks to the acts_as_tenant gem. No need to build them using active record associations directly.

Reply:
> All looks good. Raising errors in controller actions is different from implementation in other controllers in this app but I am not opposed to it.
> 
> One thing, can we use `DefaultPetTask.new` instead of `@organization.default_pet_tasks.build` because `DefaultPetTask` has a belongs_to relatinship to Organization. So any instances will be scoped to the organization automatically thanks to the acts_as_tenant gem. No need to build them using active record associations directly.

I see that this `act_as_tenant` checks the URL to set the tenant, but the routes of default pet tasks are not nested and the request can still be performed without organization as a param
![image](https://github.com/rubyforgood/pet-rescue/assets/33605808/a956be08-a15e-4356-bdc0-cf7dd5137cb4)

Then we have a security issue, I just tried to create a default pet task for another organization by manually changing the organization in the form path
![image](https://github.com/rubyforgood/pet-rescue/assets/33605808/8441fd68-4feb-4b34-8e00-c88001226c2e)
if we don't restrict organization param in the controller then this will happen
![image](https://github.com/rubyforgood/pet-rescue/assets/33605808/e165dc39-bc2e-4d49-a32e-cd019750c0fc)

I checked `OrganizationScopable`, and somehow `current_user` got lost on the way here, but the organization is being set to the one that I manually entered and a default pet task is created with the wrong organization.
![image](https://github.com/rubyforgood/pet-rescue/assets/33605808/b5fd92b6-ef08-4491-8978-e099762f3f15)

Reply:
Ooh I see. Thank you for the information! Can we just scope the routes under an organization? They should be.

Reply:
> Ooh I see. Thank you for the information! Can we just scope the routes under an organization? They should be.

I guess we can, but not as a part of this task, since it is out of scope for this particular issue as it requires changes throughout the application. For this task, I suggest leaving the issue I've discovered as it is and open another issue that to fix this vulnerability and modify/refactor routes/act_as_tenant functionality.

Reply:
> Ooh I see. Thank you for the information! Can we just scope the routes under an organization? They should be.

And I will apply changes you've requested a bit later

Reply:
> Ooh I see. Thank you for the information! I am on my phone so can't look in depth but I see that default let tasks are within the org block in the routes so not sure why it's not being scope to an org. Would we have a similar issue for other resources like Pet? ![Screenshot_20240106-100736](https://private-user-images.githubusercontent.com/95949082/294707889-8417ff6b-0f67-4f3e-ae8d-2c5a745514e0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ1NjYxNTMsIm5iZiI6MTcwNDU2NTg1MywicGF0aCI6Ii85NTk0OTA4Mi8yOTQ3MDc4ODktODQxN2ZmNmItMGY2Ny00ZjNlLWFlOGQtMmM1YTc0NTUxNGUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA2VDE4MzA1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVlYjI4NDNlOGQ5MDE1Y2VmOWZkOGQ5NzU0Yjc3ZGFkZWZiMzdkZWFjZjhhOTE4YTM5YjY3NzA5OTZhYTM0ZjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.AUN1akhhclDREim4qPPw7j8_zc7GTXB3QEHB27uRgiY)

it's not scoped because it uses `module:`
![image](https://github.com/rubyforgood/pet-rescue/assets/33605808/bf997e6c-e7e0-4e50-a231-26ad038b1518)
but even if it wasn't that won't add `organization_id` as it will just be `/organizations/default_pet_tasks/` instead of `/atla/default_pet_task`
to achieve what is intended we need to use the example I've provided above 
![image](https://github.com/rubyforgood/pet-rescue/assets/33605808/7c9b502d-906e-48ee-b6ce-7fedc11509eb)

Reply:
Yes you're right. I didn't realise this so thanks for explaining it to me. It now explains why in the org/pet controller we call `pet_in_same_organization?(org_id)` (method lives in app controller) on actions that CRUD a record. Perhaps we can implement similar here. It's not perfect, but will be consistent and something we can address at a later time as you stated, not in this task.

Reply:
@vmyts539 What do you think about my above comment? No rush! Just making sure you saw it as I did not @ you.

Reply:
Great, thank you @vmyts539 . DId you want to split that into two issues?

Reply:
> Great, thank you @vmyts539 . DId you want to split that into two issues?

Yeah, I'll finish with this one, then I'll investigate the possible changes to the application logic, and based on that, maybe we will create a task to rework stuff

Reply:
Nice find @vmyts539! 👏 
I also think our current setup with `acts_as_tenant` is sometimes problematic. The organization slug is not handled by routes but instead by a custom Middleware. This can be surprising and doesn't always play nice with rails (e.g. we it caused problems with url generation in emails).

Alto it is not perfect I think what you did with `verify_organization_for_current_user` is sufficient for now, we just need to ensure it is called before every action in the admin panel but that's a separate issue 👍